Framework/PythonInterface/test/python/mantid/utils/loggingTest.py

```diff
@@ -7,14 +7,34 @@
 
 # package imports
 from mantid.kernel import ConfigService, logger
-from mantid.utils.logging import capture_logs
+from mantid.utils.logging import capture_logs, log_to_python
 
 # standard imports
-from pathlib import Path
-import os
+from contextlib import contextmanager
+import logging
 import unittest
 
 
+class CaptureHandler(logging.Handler):
+    def __init__(self):
+        super().__init__(level=logging.DEBUG)
+        self.records = []
+
+    def emit(self, record: logging.LogRecord):
+        self.records.append(record)
+
+
+@contextmanager
+def backup_config():
```

Comment:
It feels like there is a better place to put this. I've written this same functionality myself a bunch of times which isn't very DRY.

Reply:
How about
```.py
config = ConfigService.Instance()
with config.temporary():
    config[key] = value
    # do stuff
# config gets reset here
```
But I don't know of the larger implications here. Would backing up and restoring the entire config re-init important components and cause problems?

Or are you referring to the logging config specifically?

Reply:
I meant as a utility for testing, not specific to logging. If you can think of a good place to put this so we can advertise it to developers I'd be very appreciative. However, it is outside the scope of this PR so I won't block merging based on this.

Reply:
Looks like `Framework/PythonInterface/test/testhelpers` is a good place. Could be either in `__init__.py` or in a new separate module.